Library/Homebrew/test/test_keg.rb

```diff
@@ -304,4 +304,35 @@ def test_removes_broken_symlinks_that_conflict_with_directories
     keg.unlink
     keg.uninstall
   end
+
+  def test_mach_o_files_skips_hardlinks
+    a = HOMEBREW_CELLAR.join("a", "1.0")
+    a.join("lib").mkpath
+    FileUtils.cp dylib_path("i386"), a.join("lib", "i386.dylib")
+    FileUtils.ln a.join("lib", "i386.dylib"), a.join("lib", "i386_link.dylib")
+
+    keg = Keg.new(a)
+    keg.link
+
+    assert_equal 1, keg.mach_o_files.size
+  ensure
+    keg.unlink
+    keg.uninstall
+  end
+
+  def test_mach_o_files_isnt_confused_by_symlinks
+    a = HOMEBREW_CELLAR.join("a", "1.0")
+    a.join("lib").mkpath
+    FileUtils.cp dylib_path("i386"), a.join("lib", "i386.dylib")
+    FileUtils.ln a.join("lib", "i386.dylib"), a.join("lib", "i386_link.dylib")
+    FileUtils.ln_s a.join("lib", "i386.dylib"), a.join("lib", "1.dylib")
```

Comment:
I'd personally prefer if we used `Pathname#/` instead of `Pathname#join` in all our test code (or at least the newly added one). I find it easier to read and it's closer to what we do in some (most?) of Homebrew and particularly in our formulae.